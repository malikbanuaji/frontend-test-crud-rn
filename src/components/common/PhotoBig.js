import React, {useMemo} from 'react';
import {
  Image,
  StyleSheet,
  Text,
  TouchableNativeFeedback,
  useWindowDimensions,
  View,
} from 'react-native';
import {Colors, Mixins, Typography} from '../../styles';

export default function PhotoBig({onPress, photo, firstName, lastName}) {
  const windows = useWindowDimensions();
  const photoSize = useMemo(() => {
    return {
      width: windows.width * 0.3,
      height: windows.width * 0.3,
    };
  }, [windows.width]);
  return (
    <View style={styles.photo}>
      <TouchableNativeFeedback onPress={onPress}>
        <View style={[styles.photo]}>
          {photo === 'N/A' ? (
            <View style={[photoSize, styles.emptyPhoto]}>
              <Text style={styles.emptyText}>
                {`${firstName.substring(0, 1)}${lastName.substring(
                  0,
                  1,
                )}`.toUpperCase()}
              </Text>
            </View>
          ) : (
            <Image style={photoSize} {...photoSize} source={{uri: photo}} />
          )}
        </View>
      </TouchableNativeFeedback>
    </View>
  );
}

PhotoBig.defaultProps = {
  firstName: '',
  lastName: '',
  photo: 'N/A',
};

const styles = StyleSheet.create({
  photo: {
    ...Mixins.photo,
  },
  emptyPhoto: {
    ...Mixins.emptyPhoto,
  },
  emptyText: {
    ...Typography.photoBigEmptyText,
  },
});
