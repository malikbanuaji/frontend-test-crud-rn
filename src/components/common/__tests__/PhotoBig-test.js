import React from 'react';
import renderer from 'react-test-renderer';
import PhotoBig from '../PhotoBig';

test('renders correctly with image', () => {
  const tree = renderer
    .create(
      <PhotoBig
        photo={
          'https://mattermost.com/wp-content/uploads/2018/10/React_Native_Logo.png'
        }
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly without image', () => {
  const tree = renderer.create(<PhotoBig photo={'N/A'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly without firstName', () => {
  const tree = renderer.create(<PhotoBig lastName={'Doe'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly without lastName', () => {
  const tree = renderer.create(<PhotoBig firstName={'Doe'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with full props', () => {
  const tree = renderer
    .create(
      <PhotoBig
        firstName={'Doe'}
        lastName={'Doe'}
        photo={
          'https://mattermost.com/wp-content/uploads/2018/10/React_Native_Logo.png'
        }
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
