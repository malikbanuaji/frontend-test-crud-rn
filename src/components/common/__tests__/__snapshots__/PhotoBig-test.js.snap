// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with full props 1`] = `
<View
  style={
    Object {
      "borderRadius": 8,
      "overflow": "hidden",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 8,
          "overflow": "hidden",
        },
      ]
    }
  >
    <Image
      height={225}
      source={
        Object {
          "uri": "https://mattermost.com/wp-content/uploads/2018/10/React_Native_Logo.png",
        }
      }
      style={
        Object {
          "height": 225,
          "width": 225,
        }
      }
      width={225}
    />
  </View>
</View>
`;

exports[`renders correctly with image 1`] = `
<View
  style={
    Object {
      "borderRadius": 8,
      "overflow": "hidden",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 8,
          "overflow": "hidden",
        },
      ]
    }
  >
    <Image
      height={225}
      source={
        Object {
          "uri": "https://mattermost.com/wp-content/uploads/2018/10/React_Native_Logo.png",
        }
      }
      style={
        Object {
          "height": 225,
          "width": 225,
        }
      }
      width={225}
    />
  </View>
</View>
`;

exports[`renders correctly without firstName 1`] = `
<View
  style={
    Object {
      "borderRadius": 8,
      "overflow": "hidden",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 8,
          "overflow": "hidden",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 225,
            "width": 225,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#93b5e1",
            "borderRadius": 8,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 48,
          }
        }
      >
        D
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly without image 1`] = `
<View
  style={
    Object {
      "borderRadius": 8,
      "overflow": "hidden",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 8,
          "overflow": "hidden",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 225,
            "width": 225,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#93b5e1",
            "borderRadius": 8,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 48,
          }
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly without lastName 1`] = `
<View
  style={
    Object {
      "borderRadius": 8,
      "overflow": "hidden",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 8,
          "overflow": "hidden",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 225,
            "width": 225,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#93b5e1",
            "borderRadius": 8,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 48,
          }
        }
      >
        D
      </Text>
    </View>
  </View>
</View>
`;
